astropy/table/column.py

```diff
@@ -711,6 +711,13 @@ def __new__(cls, data=None, name=None,
                                           copy=copy, copy_indices=copy_indices)
         return self
 
+    def __set_attr__(self, item):
```

Comment:
On the other hand, if a `@property` is used then it *will* look on dir() like non-masked columns have a "mask" attribute so that could be confusing after all.

On the third hand, I'm not sure this shouldn't go on `BaseColumn` since we would want this behavior for any column that *doesn't* have a `mask` attribute.

Maybe something like `if not hasattr(self, 'mask'): raise AttributeError(...)`